fix(ivy): TestBed not unwrapping imports array function when overriding provider

crisbeto opened the pull request:
Fixes an error that is thrown when a provider is overridden in `TestBed`, if the module definition of one of the imported modules uses a function for the `imports` that is set via `setNgModuleScope`. The problem was that we have a `for...of` loop that assumes that the imports are an array, but they can also be a function. This was handled correctly in other places, but the one here was missed.

Note that the above-mentioned error is only thrown at runtime when the code is transpiled to es6. In es5 TS generates a call to a helper that handles the error silently so the attached unit test only fails in es6.

Fixes #34623.

AndrewKushnir commented on packages/core/test/test_bed_spec.ts:
```diff
@@ -1002,4 +1003,49 @@ describe('TestBed', () => {
+
+            let fixture: ComponentFixture<AppComponent>;
+
+            expect(() => {
```
@crisbeto I'd propose to update test description a bit to say `should handle provider overrides when module imports are provided as a function` and remove the `expect(() => { ... }).not.toThrow()` wrapper, i.e. keep the test more generic.

crisbeto replied:
Done.

AndrewKushnir reviewed (approved):
LGTM (with one minor comment), thanks for the fix @crisbeto 👍

AndrewKushnir commented:
[Presubmit](http://test/OCL:288007601:BASE:288007631:1578074375221:38c23646)

crisbeto committed:
fix(ivy): TestBed not unwrapping imports array function when overriding provider

Fixes an error that is thrown when a provider is overridden in `TestBed`, if the module definition of one of the imported modules uses a function for the `imports` that is set via `setNgModuleScope`. The problem was that we have a `for...of` loop that assumes that the imports are an array, but they can also be a function. This was handled correctly in other places, but this one was missed.

Note that the above-mentioned error is only thrown at runtime when the code is transpiled to es6. In es5 TS generates a call to a helper that handles the error silently so the attached unit test only fails in es6.

Fixes #34623.
